array.c

```diff
@@ -5971,29 +5972,29 @@ rb_ary_diff(VALUE ary1, VALUE ary2)
 
 /*
  *  call-seq:
- *     ary.difference(other_ary1, other_ary2, ...)   -> new_ary
- *
- *  Array Difference
- *
- *  Returns a new array that is a copy of the original array, removing all
- *  occurrences of any item that also appear in +other_ary+. The order is
- *  preserved from the original array.
- *
- *  It compares elements using their #hash and #eql? methods for efficiency.
+ *    array.difference(*other_arrays) -> new_array
  *
- *     [ 1, 1, 2, 2, 3, 3, 4, 5 ].difference([ 1, 2, 4 ])     #=> [ 3, 3, 5 ]
+ *  Each argument in +other_arrays+ must be an
+ *  {Array-convertible object}[doc/implicit_conversion_rdoc.html#label-Array-Convertible+Objects].
  *
- *  Note that while 1 and 2 were only present once in the array argument, and
- *  were present twice in the receiver array, all occurrences of each Integer are
- *  removed in the returned array.
+ *  Returns a new \Array containing only those elements from +self+
+ *  that are not found in any of the +other_arrays+;
+ *  items are compared using <tt>eql?</tt>;  order from +self+ is preserved:
+ *    [0, 1, 1, 2, 1, 1, 3, 1, 1].difference([1]) # => [0, 2, 3]
+ *    [0, 1, 2, 3].difference([3, 0], [1, 3]) # => [2]
+ *    [0, 1, 2].difference([4]) # => [0, 1, 2]
  *
- *  Multiple array arguments can be supplied and all occurrences of any element
- *  in those supplied arrays that match the receiver will be removed from the
- *  returned array.
+ *  Returns a copy of +self+ if no arguments given:
```

Comment:
Not sure if this warrants explicitly documenting, it is implied by the above description.  Similar issues for other methods.

Reply:
It does say that no arguments are required.  And for intersection, one might think that an intersection with 'nothing' could return an empty Array, which is not the case.

Reply:
Fair enough.  You may want to change the examples to show a splatting of an argument that is an empty array, because these are methods that it doesn't make sense to call explicitly without arguments, unless the code is purposely trying to confuse.

Reply:
Good point.  I was not entirely comfortable with what I had, but b/c this is API doc, I want to leave nothing out.  Will modify.

Reply:
This is fixed (*[]) for the three here and for concat.